libtiledbsoma/test/test_soma_array.py

```diff
@@ -15,6 +15,35 @@
     clib.config_logging("debug")
 
 
+def test_soma_array_basic_getters():
+    name = "obs"
+    uri = os.path.join(SOMA_URI, name)
+    column_names = [
+        "soma_joinid",
+        "obs_id",
+        "n_genes",
+        "percent_mito",
+        "n_counts",
+        "louvain",
+        "is_b_cell",
+    ]
+    result_order = clib.ResultOrder.colmajor
+
+    sr = clib.SOMAArray(
+        uri=uri, name=name, column_names=column_names, result_order=result_order
+    )
+
+    assert sr.uri == uri
+    assert sr.nnz() == 2638
+
+    # Note that column_names is set AFTER read_next even though it is an arugment
```

Comment:
Would be nice-to-have if this prop was supported even before doing a read_next.  I.e., given a SOMARead, you can always get the column names regardless of read state.

I did a quick code read, and it looks like reset is called during the SOMAArray constructor.  This initial reset() call doesn't include the column names, so the ManagedQuery.  If the initial reset included the column names, would this issue be resolved?